File: src/main/java/com/phakel/event/Subscribe.java
package com.phakel.event;

import java.lang.annotation.*;

/**
 * An annotation for signing a Listener
 * @author EvanLuo42
 */
@Documented
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Subscribe {

    /**
     * @return The priority of this Listener.
     */
    Priority priority() default Priority.NORMAL;
}

File: src/main/kotlin/cn/phakel/easyeventbus/listener/Subscribe.kt
package cn.phakel.easyeventbus.listener

import cn.phakel.easyeventbus.event.Event
import kotlin.reflect.KClass

/**
 * Annotation Class for @Subscribe
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class Subscribe(val event: KClass<out Event>, val priority: ListenerPriority)